testutils/helper/curl.go

```diff
@@ -26,13 +26,21 @@ type CurlOpts struct {
 	WithoutStats bool
 }
 
-func (t *testContainer) CurlEventuallyShouldRespond(opts CurlOpts, substr string, ginkgoOffset int, timeout time.Duration) {
+func (t *testContainer) CurlEventuallyShouldRespond(opts CurlOpts, substr string, ginkgoOffset int, timeout ...time.Duration) {
 	defaultTimeout := time.Second * 20
-	if timeout == 0 {
-		timeout = defaultTimeout
+	currentTimeout := defaultTimeout
+	if len(timeout) > 0 {
```

Comment:
Nit: maybe a `switch len(timeout)` where you overwrite the defaults in each `case` clause would be more readable.